test/publisher/index.ts

```diff
@@ -347,6 +347,11 @@ describe('Publisher', () => {
         sandbox
           .stub(FakeQueue.prototype, '_publish')
           .callsFake((messages, callbacks, callback) => {
+            // Simulate the drain taking longer than the publishes. This can
+            // happen if more messages are queued during the publish().
+            setTimeout(() => {
+              publisher.queue.emit('drain');
+            }, 50);
```

Comment:
This will fail eventually, I can bet :)  Maybe use longer timeout?

Reply:
This just needs to be on the next tick, really, so I could just change this to nextTick().